app/services/hotkeys.ts

```diff
@@ -40,148 +53,179 @@ export interface IBinding {
   };
 }
 
-
 interface IHotkeyAction {
   name: string;
+
   description(entityId: string): string;
-  down(entityId: string): void;
+
+  down(entityId: string, hotkeyId?: number): void;
+
   isActive?(entityId: string): boolean;
+
   shouldApply?(entityId: string): boolean;
-  up?(entityId: string): void;
+
+  up?(entityId: string, hotkeyId?: number): void;
 
   // These are injected dynamically
   downHandler?(): void;
+
   upHandler?(): void;
 }
 
+type HotkeyGroup = {
+  [x: string]: IHotkeyAction;
```

Comment:
what is `x`? I assume it must be `actionName` here

Reply:
This is conventional indexer syntax in TS, do you think having `actionName` would really increase readability?

Reply:
yes

Reply:
Changed